polyfill/lib/ecmascript.mjs

```diff
@@ -3378,7 +3413,7 @@ export const ES = ObjectAssign({}, ES2020, {
         remainder = years;
```

Comment:
Is an O(n) loop necessary for years?  If there's a 400K-year duration, that's a lot of looping! Although some calendars may have a variable number of years, ISO years (the vast majority of usage) are deterministic and this calculation could probably be O(1) for ISO.  Looping for days/months seems OK because there's not going to be 1000s of them.

Reply:
I was also thinking over the weekend that a lot of this code could be skipped in many cases if we used getNextTransition/getPreviousTransition.

Would you be interested in doing the walkthrough and possibly identifying ways to optimize these algorithms next week? I'm not sure I can bring myself to look at them again this week. :grimacing:

Reply:
Sure, would be happy to review next week!  I spent today doing a deep dive on the new code and have two questions that may be bugs in the algorithm. Will make a separate comment about that to get your feedback. 

> I was also thinking over the weekend that a lot of this code could be skipped in many cases if we used getNextTransition/getPreviousTransition.

Some time zones may not implement these functions-- they're optional AFAIK. Seems like a better approach might be to use PlainDateTime math and then adjust the endpoint as needed.  But let's discuss when we chat 1:1 next week.